File: templates/Julia/tmp.jl
#!/usr/bin/env -C julia -t auto

# Define a UserData struct to hold user information
struct UserData
    name::String
    age::UInt8
end

"""
    validate_age(user::UserData)

Validate the age of a user.

# Arguments
- `user::UserData`: A UserData instance whose age needs to be validated.

# Returns
- `Tuple{Union{Nothing, UserData}, Union{Nothing, String}}`: 
  Returns `(user, nothing)` if age is valid, else returns `(nothing, "error message")`.

"""
function validate_age(user::UserData)
    if user.age < 18
        return nothing, "User $(user.name) is under 18."
    elseif user.age > 120
        return nothing, "User $(user.name)'s age $(user.age) is not valid."
    else
        return user, nothing
    end
end
precompile(validate_age, (UserData,))

function main()
    # Create UserData instances
    user1 = UserData("Alice", 25)
    user2 = UserData("Bob", 15)
    user3 = UserData("Charlie", 130)

    # Create an array of users
    users = [user1, user2, user3]

    # Validate each user
    for user in users
        valid_user, err = validate_age(user)
        if err === nothing
            println("User $(valid_user.name) has a valid age.")
        else
            println(err)
        end
    end
end
precompile(main)

# Run the main function
main()

File: templates/Go/tmp.go
package main

import (
	"errors"
	"fmt"
)

// UserData struct to hold user information
type UserData struct {
	name string
	age  uint8
}

// ValidateAge function to validate user age
func ValidateAge(user *UserData) (*UserData, error) {
	if user.age < 18 {
		return nil, errors.New(fmt.Sprintf("User %s is under 18.", user.name))
	} else if user.age > 120 {
		return nil, errors.New(fmt.Sprintf("User %s's age %d is not valid.", user.name, user.age))
	} else {
		return user, nil
	}
}

func main() {
	// Create UserData instances
	user1 := &UserData{name: "Alice", age: 25}
	user2 := &UserData{name: "Bob", age: 15}
	user3 := &UserData{name: "Charlie", age: 130}

	// Create a slice of users
	users := []*UserData{user1, user2, user3}

	// Validate each user
	for _, user := range users {
		validUser, err := ValidateAge(user)
		if err != nil {
			fmt.Println(err)
		} else {
			fmt.Printf("User %s has a valid age.\n", validUser.name)
		}
	}
}
